packages/web3-common/src/eth_execution_api.ts

```diff
@@ -0,0 +1,293 @@
+		readonly userDoc: {
+			readonly methods: Record<string, unknown>;
+		};
+		readonly developerDoc: {
+			readonly methods: Record<string, unknown>;
+		};
+	};
+}
+
+/* eslint-disable camelcase */
+export type EthExecutionAPI = {
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/block.json
+	eth_getBlockByHash: (blockHash: HexString32Bytes, hydrated: boolean) => Block;
+	eth_getBlockByNumber: (blockNumber: BlockNumberOrTag, hydrated: boolean) => Block;
+	eth_getBlockTransactionCountByHash: (blockHash: HexString32Bytes) => Uint;
+	eth_getBlockTransactionCountByNumber: (blockNumber: BlockNumberOrTag) => Uint;
+	eth_getUncleCountByBlockHash: (blockHash: HexString32Bytes) => Uint;
+	eth_getUncleCountByBlockNumber: (blockNumber: Uint) => Uint;
+	eth_getUncleByBlockHashAndIndex: (blockNumber: BlockNumberOrTag, uncleIndex: Uint) => Block;
+	eth_getUncleByBlockNumberAndIndex: (blockNumber: BlockNumberOrTag, uncleIndex: Uint) => Block;
+
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/transaction.json
+	eth_getTransactionByHash: (transactionHash: HexString32Bytes) => TransactionInfo;
+	eth_getTransactionByBlockHashAndIndex: (
+		blockHash: HexString32Bytes,
+		transactionIndex: Uint,
+	) => TransactionInfo;
+	eth_getTransactionByBlockNumberAndIndex: (
+		blockNumber: BlockNumberOrTag,
+		transactionIndex: Uint,
+	) => TransactionInfo;
+	eth_getTransactionReceipt: (transactionHash: HexString32Bytes) => ReceiptInfo;
+
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/client.json
+	eth_protocolVersion: () => string;
+	eth_syncing: () => SyncingStatus;
+	eth_coinbase: () => Address;
+	eth_accounts: () => Address[];
+	eth_blockNumber: () => Uint;
+
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/execute.json
+	eth_call: (transaction: TransactionCall, blockNumber: BlockNumberOrTag) => HexStringBytes;
+	eth_estimateGas: (transaction: TransactionWithSender, blockNumber: BlockNumberOrTag) => Uint;
+
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/fee_market.json
+	eth_gasPrice: () => Uint;
+	eth_feeHistory: (
+		blockCount: Uint,
+		newestBlock: BlockNumberOrTag,
+		rewardPercentiles: number[],
+	) => FeeHistoryResult;
+
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/filter.json
+	eth_newFilter: (filter: Filter) => Uint;
+	eth_newBlockFilter: () => Uint;
+	eth_newPendingTransactionFilter: () => Uint;
+	eth_uninstallFilter: (filterIdentifier: Uint) => boolean;
+	eth_getFilterChanges: (filterIdentifier: Uint) => FilterResults;
+	eth_getFilterLogs: (filterIdentifier: Uint) => FilterResults;
+	eth_getLogs: (filter: Filter) => FilterResults;
+
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/mining.json
+	eth_mining: () => boolean;
+	eth_hashrate: () => Uint;
+	eth_getWork: () => [HexString32Bytes, HexString32Bytes, HexString32Bytes];
+	eth_submitWork: (
+		powHash: HexString32Bytes,
+		seedHash: HexString32Bytes,
+		difficulty: HexString32Bytes,
```

Comment:
These might be incorrect, created an [issue](https://github.com/ethereum/execution-apis/issues/119)